selfdrive/boardd/boardd.cc

```diff
@@ -884,60 +650,24 @@ int main() {
     fake_send = true;
   }
 
-  if (getenv("BOARDD_LOOPBACK")){
-    loopback_can = true;
-  }
+  std::thread health_thread(can_health_thread);
 
-  err = pthread_mutex_init(&usb_lock, NULL);
-  assert(err == 0);
+  while (!do_exit){
```

Comment:
Why the while loop? We do the version check in pandad, what if openpilot starts -> pandad upates panda -> pandad launches boardd -> panda is disconnected -> different panda with old firmware is connected?

Reply:
This is what we've always done. I agree that maybe just exiting and having pandad restart boardd might be more logical. I'll leave behavior the same as before the refactor, but we can revisit this later.

Reply:
Once this is a more complete panda library, we should be able to move all the pandad functionality into boardd.